packages/bitcore-wallet-service/src/lib/chain/eth/index.ts

```diff
@@ -534,29 +534,39 @@ export class EthChain implements IChain {
     let multisigContractAddress;
     let address;
     let amount;
-    if (tx.abiType && tx.abiType.type === 'ERC20') {
-      tokenAddress = tx.to;
-      address = Web3.utils.toChecksumAddress(tx.abiType.params[0].value);
-      amount = tx.abiType.params[1].value;
-    } else if (tx.abiType && tx.abiType.type === 'MULTISIG' && tx.abiType.name === 'submitTransaction') {
-      multisigContractAddress = tx.to;
-      address = Web3.utils.toChecksumAddress(tx.abiType.params[0].value);
-      amount = tx.abiType.params[1].value;
-    } else if (tx.abiType && tx.abiType.type === 'MULTISIG' && tx.abiType.name === 'confirmTransaction') {
-      multisigContractAddress = tx.to;
-      address = '0x0';
-      amount = 0;
-      if (tx.internal && tx.internal.length > 0) {
-        address = Web3.utils.toChecksumAddress(tx.internal[0].action.to);
-        amount = tx.internal[0].action.value;
-      } else if (tx.calls && tx.calls.length > 0) {
-        address = Web3.utils.toChecksumAddress(tx.calls[0].to);
-        amount = tx.calls[0].value;
+    // Only returns single notification so determine most precise
+    if (tx.effects && tx.effects.length) {
+      const multisigConfirm = tx.effects.findIndex(e => e.type === 'MULTISIG:confirmTransaction');
+      const multisigSubmit = tx.effects.findIndex(e => e.type === 'MULTISIG:submitTransaction');
+      const erc20Transfer = tx.effects.findIndex(e => e.type === 'ERC20:transer');
+      const internalTransfer = tx.effects.findIndex(e => !e.type && !e.contractAddress);
+      
+      const exists = [multisigConfirm, multisigSubmit, erc20Transfer, internalTransfer].filter(i => i > -1);
+      
+      if (exists.length) {
+        // Get the first effect based on priority as determined by order in above array
+        const best = exists[0];
```

Comment:
Also, I am not able to hit this logic. `tx.effects` is always undefined.

...Turns out this never gets hit after the tx is mined, so calls/internal/effects will never be populated at this point. :shrug:

Reply:
Also, I had to do the following to get bcmonitor working locally. It might be worth adding.

![image](https://github.com/bitpay/bitcore/assets/32603939/13fe362a-2b4b-46e6-af22-6e68a18f3701)